Add automated release workflow (#875)
Parsely/wp-parsely: .github/ISSUE_TEMPLATE/release-template.md

> :warning: DO NOT MERGE (YET) :warning:

Reviewer: Maybe we could remove this line?

Author: Done.

[Remaining work for this Milestone](https://github.com/Parsely/wp-parsely/milestone/xx)

PR for tracking changes for the X.Y.Z release. Target release date: DOW DD MMMM YYYY.

## Before releasing

- [ ] Notify stakeholders of an upcoming release.
- [ ] Merge any outstanding PRs due for this release to `develop`.

> - [ ] Verify that all PRs have a `Changelog` tag.

Reviewer: ```suggestion
- [ ] Verify that all PRs in the release have a `Changelog` tag.
```

## Release proccess

**Create the release PR**

> - [ ] Create a PR that merges `develop` into `trunk`, named _Release x.y.z_.

Reviewer: Food for thought: could we add a template for the description of this kind of PRs?

Author: I think we could, given that texts are pretty standard. Do you want me to create an issue for it?

Author: Maybe it would also be possible to integrate this into the workflow.

Reviewer: We can create an issue and integrate it in the workflow in the future.

> - [ ] Merge the PR into trunk.

Reviewer: ```suggestion
- [ ] Have the PR reviewed and merge the PR into `trunk`.
```

  > This will create a new release draft. It might take some seconds to appear.

**Promote the release draft to a pre-release**
- [ ] In the release draft page, check the version numbers and changelog data.

> - [ ] Make any adjustments if needed.

Reviewer: Can we add a link to https://semver.org/ somewhere here explaining the logic of numbering?

Reviewer: Also, let's explain what's the script's default logic to bump versions.

Author: Being already lengthy as it is, I'd rather only focus on steps in this document and not mix it with explanations. However, I do see the point of having this info and we should document it somewhere.

However, I can do it if you think it would be better to have the information here.

- [ ] Save the draft as a pre-release.
  > This will commit the version numbers and changelog into `trunk` as 2 distinct commits. Changes might take a minute to appear.

**Promote the pre-release to a release**
- [ ] Go to `trunk` and inspect the version and changelog commits for correctness.
  > If commits happen to be incorrect, please stop and contact the development team.
- [ ] Go to the pre-release and promote it to a release.
  > This will merge `trunk` into `develop` and also update the `release's tag`. Changes might take a minute to appear.
- [ ] Inspect the `develop` branch and the `release's tag` to verify that they contain the version and changelog updates.
  > If this is not the case,please stop and contact the development team.

**Deploy the release to wordpress.org**
- [ ] Manually run the `Deploy to WordPress.org` GitHub workflow.
  > This will deploy the new version of the plugin to wordpress.org.

## After releasing

**Managing milestones**
- [ ] Close the current milestone.
- [ ] If needed, open a new milestone for the next release.

> - [ ] If there are any open PRs/issues that did do not make it into the release, update their milestone.

Reviewer: Should this step go at the beginning?

Author: Milestone steps can very well be moved to the beginning, as long as milestone steps 1 and 2 are sequential.

**Communicating and releasing to other platforms**
- [ ] Write an internal P2 post about the release.

> - [ ] Release the plugin for WPVIP and write a release lobby post.

Reviewer: ```suggestion
- [ ] Release the plugin for WordPress VIP and write a release lobby post.
```

> - [ ] Release the plugin for WPCOM.

Reviewer: ```suggestion
- [ ] Release the plugin for WordPress.com.
```
